File: packages/square-mobile-payments/android/src/main/java/io/capawesome/capacitorjs/plugins/squaremobilepayments/classes/results/MoneyResult.java
package io.capawesome.capacitorjs.plugins.squaremobilepayments.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.squaremobilepayments.interfaces.Result;

public class MoneyResult implements Result {

    private final int amount;

    @NonNull
    private final String currency;

    public MoneyResult(int amount, @NonNull String currency) {
        this.amount = amount;
        this.currency = currency;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("amount", amount);
        result.put("currency", currency);
        return result;
    }
}

File: packages/square-mobile-payments/ios/Plugin/Classes/Results/MoneyResult.swift
import Foundation
import Capacitor

@objc public class MoneyResult: NSObject, Result {
    let amount: Int
    let currency: String

    init(amount: Int, currency: String) {
        self.amount = amount
        self.currency = currency
    }

    @objc public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["amount"] = amount
        result["currency"] = currency
        return result as AnyObject
    }
}
